File: Seminar/src/main/java/org/sopt/Seminar/domain/post/model/Post.java
package org.sopt.Seminar.domain.post.model;

import static jakarta.persistence.FetchType.*;
import static jakarta.persistence.GenerationType.*;
import static lombok.AccessLevel.*;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.sopt.Seminar.domain.category.model.CategoryId;
import org.sopt.Seminar.domain.member.model.Member;
import org.sopt.Seminar.global.common.model.BaseTimeEntity;

@Entity
@Builder
@Getter
@NoArgsConstructor(access = PROTECTED)
@AllArgsConstructor
public class Post extends BaseTimeEntity {

    @Id @GeneratedValue(strategy = IDENTITY)
    @Column(name = "post_id")
    public Long id;

    private String title;

    @Column(columnDefinition = "TEXT")
    private String content;

    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "member_id")
    private Member member;

    @Column(name = "category_id")
    private CategoryId categoryId;

    private String imageUrl;

    @Builder(builderMethodName = "builderWithImageUrl")
    public Post(String title, String content, String imageUrl, Member member) {
        this.title = title;
        this.content = content;
        this.imageUrl = imageUrl;
        this.member = member;
    }

    public void updateContent(String content) {
        this.content = content;
    }
}

File: kopring/week2/kotlin-semina/src/main/kotlin/org/sopt/kotlinsemina/domain/post/model/Post.kt
package org.sopt.kotlinsemina.domain.post.model

import jakarta.persistence.*
import jakarta.persistence.FetchType.*
import jakarta.persistence.GenerationType.*
import org.sopt.kotlinsemina.domain.category.model.CategoryId
import org.sopt.kotlinsemina.domain.member.model.Member
import org.sopt.kotlinsemina.global.common.model.BaseTimeEntity

@Entity
class Post(
    title: String,

    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "member_id")
    val member: Member,

    @Column(name = "category_id")
    val categoryId: CategoryId,

    content: String
    ): BaseTimeEntity() {
    @Id @GeneratedValue(strategy = IDENTITY)
    @Column(name = "post_id")
    val id: Long = 0

    var title: String = title
        private set

    @Column(columnDefinition = "TEXT")
    var content: String = content
        private set

    fun updateContent(content: String) {
        this.content = content
    }
}